### PR TITLE
Fix a number of CodeQL warnings

### PR DESCRIPTION
**Pull request by dbussink**
This addresses a number of CodeQL warnings. We have a concurrency field which is using inconsistent types across different fields, which means we end up casting and triggering warnings around overflows. We can use the smallest type here, namely int32 since that is for sure sufficient for real concurrency values. It avoids warnings like:

https://github.com/vitessio/vitess/security/code-scanning/267

Also fixes the parsing bug identified in
https://github.com/vitessio/vitess/security/code-scanning/266 where we parse with `ParseUint` instead of `ParseInt`.

Last one is
https://github.com/vitessio/vitess/security/code-scanning/268 which is fixed by making sure we parse it as the type MySQL sends down and to keep using that consistently.

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

File: .github/workflows/codeql_analysis.yml [hunk 1]
```diff
@@ -29,6 +29,11 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v3
 
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.21.5
+
```
Comment (dbussink) [inline .github/workflows/codeql_analysis.yml:36]: Moved this, see the warning at https://github.com/vitessio/vitess/security/code-scanning/tools/CodeQL/status/configurations/actions-FZTWS5DIOVRC653POJVWM3DPO5ZS6Y3PMRSXC3C7MFXGC3DZONUXGLTZNVWA/a24096b7981d3f05871d576ae64dfb71efbcb9d66535f61fa746cea1eb2c8e12

**Review by frouioui — APPROVED** [review]
Looks good to me! Thank you for fixing those @dbussink 🙏🏻

File: go/vt/vtadmin/http/request.go [hunk 2]
```diff
@@ -77,6 +77,25 @@ func (r Request) ParseQueryParamAsUint32(name string, defaultVal uint32) (uint32
 	return defaultVal, nil
 }
 
+// ParseQueryParamAsInt32 attempts to parse the query parameter of the given
+// name into a uint32 value. If the parameter is not set, the provided default
+// value is returned.
+func (r Request) ParseQueryParamAsInt32(name string, defaultVal int32) (int32, error) {
+	if param := r.URL.Query().Get(name); param != "" {
```
Comment (shlomi-noach) [inline go/vt/vtadmin/http/request.go:84]: Styling-wise, I'd prefer to first handle the case where `param == ""` and do an early return, then unindent the rest of the function. But it's insignificant and not worth the rebuild time.

**Review by shlomi-noach — APPROVED** [review]

File: go/vt/vtadmin/http/request.go (hunk 2 above)
Comment (dbussink) [reply go/vt/vtadmin/http/request.go:84]: @shlomi-noach Copy-pasted this mostly from another function for `uint32`, so it's not "new" in that sense.

**Pull request merged**